tests/test_trainable_tokens.py

```diff
@@ -533,18 +588,55 @@ def model_weight_tied(self, model_id_weight_tied):
             ),
         ],
     )
-    def test_weight_tying_raises_when_detected_combined(self, model_weight_tied, peft_config):
-        # since weight tying is currently not supported make sure that an error is raised when attempting
-        # to use a model that has tied input/output embeddings
+    def test_weight_tying_noop_when_model_is_untied(self, model_weight_untied, peft_config, tmp_path):
+        # test if the weight tying is affected as well when we modified the embedding.
+        assert model_weight_untied._tied_weights_keys
+        assert not model_weight_untied.config.tie_word_embeddings
+
+        peft_model = get_peft_model(model_weight_untied, peft_config)
+        assert hasattr(peft_model.model.model.embed_tokens, "token_adapter")
+        assert not hasattr(peft_model.model.lm_head, "token_adapter")
+
+    @pytest.mark.parametrize(
+        "peft_config",
+        [
+            LoraConfig(
+                target_modules="all-linear",
+                trainable_token_indices={"embed_tokens": [0, 1, 3]},
+            ),
+        ],
+    )
+    def test_weight_tying_applied_when_model_is_tied(self, model_weight_tied, peft_config, tmp_path):
+        # test if the weight tying is affected as well when we modified the embedding.
         assert model_weight_tied._tied_weights_keys
         assert model_weight_tied.config.tie_word_embeddings
 
-        with pytest.raises(ValueError) as e:
-            peft_model = get_peft_model(model_weight_tied, peft_config)
+        peft_model = get_peft_model(model_weight_tied, peft_config)
+
+        # make it so that the input embeddings diverge. when the weights are tied this should
+        # reflect in the output embeddings as well.
+        self.simulate_training(peft_model.model.model.decoder.embed_tokens.token_adapter)
+
+        # we have to find out if the input embedding tying is doing its job during forward.
+        # for this we can leverage the fact that  emb_out(1/emb_in(x))  is  embed_dim  on the
+        # diagonal iff  emb_in.weight == emb_out.weight.
```

Comment:
Oh, hmm, okay, oh, ...

Also:
```suggestion
        # diagonal iff emb_in.weight == emb_out.weight.
```